docker/service.go

```diff
@@ -300,14 +278,28 @@ func (s *Service) up(ctx context.Context, imageName string, create bool, options
 		containers = []*Container{c}
 	}
 
-	return s.eachContainer(ctx, containers, func(c *Container) error {
-		if create {
-			if err := s.recreateIfNeeded(ctx, imageName, c, options.NoRecreate, options.ForceRecreate); err != nil {
+	hasBeenRecreate := false
+
+	return s.watchContainers(ctx, containers, func(c *Container) error {
```

Comment:
Not sure if its only me 😕 , But such compounded statement gives me hard time to navigate through code. Humble request if we can break into simple return statement and function definitions 😌